### DISCUSSION TITLE
What is SSIM's output range?

### DISCUSSION THREAD

**Discussion by ljeonjko** [francois-rozet/piqa#31] [Q&A] [answered]
Hi!
I'm using your implementation of SSIM as a loss function. I need to know the output range: is it [0,1] or [-1,1]?
Cheers!

**Comment by ljeonjko** [comment]
Found it in previous discussions. It's [0,1], meaning the loss function can be implemented as `1 - ssim(x,y)`.

**Comment by francois-rozet** [comment, marked as answer]
Hello @ljeonjko 👋,

I had to think about it, but the output range of SSIM is `[-1, 1]`. Indeed, for the first term, if `x` and `y` are positives, the range is `[0, 1]`, but the contrast sensitivity (CS) is some kind of correlation between `x` and `y`, which means it ranges from `-1` to `1`.

<img width="562" alt="image" src="https://user-images.githubusercontent.com/37352336/166906304-c08ce501-9f16-4c05-8068-045d6acca6d8.png">

However, this does not change anything if you want a loss function. Any function of the type `a - b * ssim(x, y)`, with `b` strictly positive, would have gradients in the correct direction.

> P.S. I could not find the previous discussion where this was mentioned. However, it is written in the README that the output range is `[0, 1]` which I will fix shortly.
